workspaces/rbac/plugins/rbac-backend/src/file-permissions/csv-file-watcher.ts

```diff
@@ -242,24 +236,27 @@ export class CSVFileWatcher extends AbstractFileWatcher<string[][]> {
    * addPermissionPolicies will add the new permission policies that are present in the CSV file.
    */
   private async addPermissionPolicies(): Promise<void> {
+    const meta = {
+      policies: this.csvFilePolicies.addedPolicies,
+      source: 'csv-file',
+    };
+    const auditorEvent = await this.auditor.createEvent({
+      eventId: PermissionEvents.POLICY_CREATE,
+      severityLevel: 'medium',
+      meta: { source: meta.source },
+    });
+
     try {
       await this.enforcer.addPolicies(this.csvFilePolicies.addedPolicies);
-
-      await this.auditLogger.auditLog<PermissionAuditInfo>({
-        actorId: RBAC_BACKEND,
-        message: `Created policies`,
-        eventName: PermissionEvents.CREATE_POLICY,
-        metadata: {
-          policies: this.csvFilePolicies.addedPolicies,
-          source: 'csv-file',
-        },
-        stage: HANDLE_RBAC_DATA_STAGE,
-        status: 'succeeded',
-      });
+      await auditorEvent.success({ meta });
     } catch (e) {
       this.logger.warn(
         `Failed to add or update policies ${this.csvFilePolicies.addedPolicies} after modification ${this.filePath}. Cause: ${e}`,
       );
+      await auditorEvent.fail({
```

Comment:
If you want to use the audit logger, then it makes sense to remove the logger warning above to prevent log duplication
    
      this.logger.warn(
        `Failed to add or update policies ${this.csvFilePolicies.addedPolicies} after modification ${this.filePath}. Cause: ${e}`,
      );
      
P.S.: new audit logger doesn't provide ability to log warnings unfortunately.